src/__tests__/user.spec.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable consistent-return */
+/* eslint-disable node/no-unpublished-require */
+const supertest = require('supertest');
+const app = require('../app');
+const UserModel = require('../models/user');
+const connectDatabase = require('../db/connection');
+
+const createCurrentUser = {
+  firstName: 'adnan',
+  lastName: 'khaldar',
+  email: 'adnan@outlook.com',
+  username: 'adnan',
+  provider: 'Local',
+  providerId: 'Local',
+  passwordHash: '$2b$10$R5NUgaHK51jYdi59ncmwue/lorlCHturbAmFxJ02cS38eumzNSx7O',
+};
+
+const createRandomUser = {
+  firstName: 'john',
+  lastName: 'doe',
+  email: 'john-doe@outlook.co.uk',
+  username: 'john-doe',
+  provider: 'Local',
+  providerId: 'Local',
+};
+
+let token;
+let currentUserId;
+let validUserId;
+const invalidUserId = '62ee5c280262c62f56124dfd';
+
+beforeAll(async () => {
+  connectDatabase();
+  const newUser0 = await UserModel.create(createCurrentUser);
+  const newUser1 = await UserModel.create(createRandomUser);
+
+  currentUserId = newUser0._id;
+  validUserId = newUser1._id;
+});
+
+describe('GET /api/user/:id', () => {
+  it('should return a token to be used in the next tests ', (done) => {
```

Comment:
A test describtion should be returning a token to be used for next tests, for these kind of situations, u can write a function that does this, if u are testing if the api returns a token then the test name should be like this
`Should return a token when correct user is logged in`

Reply:
The test case _is_ to test a login: the only way for the user to have a token. Still, as you pointed it out, there might be better ways, but tbh I'm not really familiar with the concept of tokens, let alone having a function to generate them. Could you please elaborate?

Reply:
Fixed. Now ready for a merge @halitbatur.